.github/workflows/e2e-k8s.yml

```diff
@@ -0,0 +1,145 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+on:
+  pull_request:
+  push:
+    branches:
+      - dev
+
+name: E2E-K8S
+
+concurrency:
+  group: E2E-K8S-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  paths-filter:
+    name: E2E-K8S-Path-Filter
+    runs-on: ubuntu-latest
+    outputs:
+      not-ignore: ${{ steps.filter.outputs.not-ignore }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
+        id: filter
+        with:
+          filters: |
+            not-ignore:
+              - '!(docs/**)'
+  e2e-k8s:
+    name: E2E-K8S-Execute
+    needs: paths-filter
+    if: ${{ (needs.paths-filter.outputs.not-ignore == 'true') || (github.event_name == 'push') }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Build Image
+        run: |
+          ./mvnw -B clean package \
+          -Dmaven.test.skip \
+          -Dmaven.javadoc.skip \
+          -Dspotless.skip=true \
+          -Dmaven.checkstyle.skip \
+          -Dmaven.deploy.skip \
+          -Ddocker.push.skip=true \
+          -Pdocker,release -Ddocker.tag=ci \
+          -pl org.apache.dolphinscheduler:dolphinscheduler-alert-server \
+          -pl dolphinscheduler-tools \
+          -pl dolphinscheduler-api \
+          -pl dolphinscheduler-master \
+          -pl dolphinscheduler-worker -am
+      - name: Create k8s Kind Cluster
```

Comment:
It seems that full k8s is not very sufficient for CI's small machines. Can we try to use some ccaled-down version of k8s such as k3s or minikube?

Reply:
`Kind` itself does not consume too many resources. I delayed the health check of the `DS-related` components mainly because multiple Java applications started simultaneously consume a lot of resources. The use of `Kind` is mainly for convenience, so that developers can easily reproduce the environment.

Reply:
I analyzed the time consumption from [here](https://github.com/apache/dolphinscheduler/actions/runs/5665279610/job/15349865668?pr=14555) and found that building the image took 7 minutes. Starting Kind and loading the image took 2 minutes. Installing and waiting for DS ready took 10 minutes. Perhaps we can split the image building into a separate task, which would also facilitate executing k8s task-related e2e tasks on the k8s cluster later.

WDYT? @SbloodyS @EricGao888

Reply:
I'm ok with it. How about 20 min for build and 20 min for e2e-k8s?

Reply:
I'm not sure about this. If different tasks run on different test machines, is there a way for them to share the images?

Reply:
I made some optimizations based on @EricGao888 's suggestions, and the time consumption has been greatly reduced. As for splitting the tasks, perhaps we can do it when adding k8s task-related e2e test cases? @SbloodyS

Reply:
> I made some optimizations based on @EricGao888 's suggestions, and the time consumption has been greatly reduced. As for splitting the tasks, perhaps we can do it when adding k8s task-related e2e test cases? @SbloodyS

Sure.